helion/_compiler/tile_strategy.py

```diff
@@ -670,11 +670,19 @@ def codegen_device_loop(self, state: CodegenState) -> DeviceLoopState:
                 type_comment=None,
             )
             assert for_node.body is body
-            extra_body = [
-                statement_from_string(
-                    f"{index_var} = {offset_var} + tl.arange(0, ({block_size_var})).to({dtype})"
-                ),
-            ]
+            extra_body = []
+            if block_size == 1:
+                extra_body.append(
+                    statement_from_string(
+                        f"{index_var} = {offset_var} + tl.zeros([1], {dtype})"
```

Comment:
Doesn't this do the same thing as arange?  I'd expect we wuld need `shape=[]` or even just offset_var directly?

Reply:
Yes, this does the same thing. We don't need to make this change to fix tile indexing when `block_size=1`. 
However, why does `grid_codegen` handle block_size == 1 differently with `tl.zeros` instead `tl.arange`?